base/abstractarray.jl

```diff
@@ -149,7 +149,36 @@ valtype(A::Type{<:AbstractArray}) = eltype(A)
 prevind(::AbstractArray, i::Integer) = Int(i)-1
 nextind(::AbstractArray, i::Integer) = Int(i)+1
 
+
+"""
+    eltype(type)
+
+Determine the type of the elements generated by iterating a collection of the given `type`.
+For dictionary types, this will be a `Pair{KeyType,ValType}`. The definition
+`eltype(x) = eltype(typeof(x))` is provided for convenience so that instances can be passed
+instead of types. However the form that accepts a type argument should be defined for new
+types.
+
+# Examples
+```jldoctest
+julia> eltype(fill(1f0, (2,2)))
+Float32
+
+julia> eltype(fill(0x1, (2,2)))
+UInt8
+```
+"""
+eltype(::Type) = Any
+eltype(::Type{Bottom}) = throw(ArgumentError("Union{} does not have elements"))
+eltype(x) = eltype(typeof(x))
 eltype(::Type{<:AbstractArray{E}}) where {E} = @isdefined(E) ? E : Any
+
+"""
+    elsize(type)
+
+Compute the stride between consecutive elements of `eltype` stored inside the
+given `type`, if the array elements are stored densely with a uniform stride.
+"""
```

Comment:
I think it's confusing to mention "stride" here since it's not related to `stride` or `strides`. Maybe "The space taken up by a single element, including alignment padding?"

Reply:
It's related in that its usage is as the multiplication factor for converting `strides` from elements to bytes:
```
$ grep -RI elsize base
base/abstractarray.jl:_memory_offset(x::DenseArray, I::Vararg{Any,N}) where {N} = (_to_linear_index(x, I...) - first(LinearIndices(x)))*elsize(x)
base/abstractarray.jl:    return sum(map((i, s, o)->s*(i-o), J, strides(x), Tuple(first(CartesianIndices(x)))))*elsize(x)
```